namespace-lock.go

```diff
@@ -30,37 +76,36 @@ type nsParam struct {
 
 // nsLock - provides primitives for locking critical namespace regions.
 type nsLock struct {
-	*dsync.DRWMutex
-	ref uint
+	rwlock RWLocker
+	ref    uint
 }
 
 // nsLockMap - namespace lock map, provides primitives to Lock,
 // Unlock, RLock and RUnlock.
 type nsLockMap struct {
+	isDist  bool
```

Comment:
Let's document the fields here this time. We need to specify what 'isDir' stands for.